### PR TITLE
Initial implementation of IBC Precompile (branch forked from kava/release/1.10)

### PR DESCRIPTION
**Pull request by evgeniy-scherbina**
Moved to https://github.com/Kava-Labs/go-ethereum/pull/17

### CODE REVIEW

File: accounts/abi/abi.go [hunk 1]
```diff
@@ -81,6 +81,26 @@ func (abi ABI) Pack(name string, args ...interface{}) ([]byte, error) {
 	return append(method.ID, arguments...), nil
 }
 
+// getInputs gets input arguments of the given [name] method.
+func (abi ABI) getInputs(name string, data []byte, useStrictMode bool) (Arguments, error) {
```
Comment (nddeluca) [inline accounts/abi/abi.go]: These don't depend on non-private methods or data -- let's not extend the abi type and use composition or create helpers in the kava repo

File: core/vm/interface.go [hunk 2]
```diff
@@ -74,6 +76,9 @@ type StateDB interface {
 	AddPreimage(common.Hash, []byte)
 
 	ForEachStorage(common.Address, func(common.Hash, common.Hash) bool) error
+
+	Context() context.Context
+	IBCTransfer(goCtx context.Context, msg *contract.MsgTransfer) (*contract.MsgTransferResponse, error)
```
Comment (nddeluca) [inline core/vm/interface.go]: See #15, lets use a type assertion route and avoid modifying these core types

**Review by nddeluca — CHANGES_REQUESTED** [review]
Good spike but let's avoid extending core types & code.  This is a bit coupled to cosmos/kava and causes quite a bit of duplication.

File: accounts/abi/abi.go (hunk 1 above)
Comment (evgeniy-scherbina) [reply accounts/abi/abi.go]: We agreed to left code as it is, doc comments were added here: 1da08d0345d3ad0f24116b0bda0ee9c1c932f33d

Comment (nddeluca) [reply accounts/abi/abi.go]: Thnx for adding comments -- see the symmetry now and that extending the type is a better path

File: core/vm/interface.go (hunk 2 above)
Comment (evgeniy-scherbina) [reply core/vm/interface.go]: Fixed here: a66d490d3d3b061c91bbe90772e3a1c10e60a9f8

Related changes:
- ethermint: https://github.com/Kava-Labs/ethermint/pull/37/commits/638be0d0efe93866ce9d2e1a4c37262ab9cf7f93
- kava: https://github.com/Kava-Labs/kava/pull/1826/commits/11a2b8ed0f56124225839d9612abf07afe1e3a5f

**Pull request closed without merging**